### PR TITLE
Export CSV: Home Office

### PR DESCRIPTION
**Pull request by padv2010**
## Trello Card Link

https://trello.com/c/ZHLorhH6/126-service-settings-download-csv-for-ho

## Description

Download the `CSV` for Home Office according with the following details:
- URN
- Forename
- Surname
- DOB
- Nationality
- Passport Number
- Visa Type
- Date of UK entry

## Pending

I am not tracking when a report has been downloaded in order to filter out applicants
based on that date. This requirements needs further discussions with the business
in case we need to handle it.

## Technical notes

I have isolated the reports (name and data) in an object that is pulled out in the controller
based on the parameter `id`.

```ruby
def show
  report = find_report(params[:id])

  send_data(report.csv, filename: report.name)
end

private

def find_report(report_id)
  {
    home_office: Reports::HomeOffice.new,
  }.with_indifferent_access.fetch(report_id)
end
```

## Screenshot

image

### CODE REVIEW

**Comment by padv2010** [comment]
[Service Settings: Download CSV for HO](https://trello.com/c/ZHLorhH6/126-service-settings-download-csv-for-ho)

File: app/models/reports/home_office.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Reports
+  class HomeOffice
+    def name
+      "Home-Office-Report.csv"
```
Comment (raul-gracia) [inline app/models/reports/home_office.rb:6]: I would include here the current date and time on the file name so its clear when the file was downloaded.

File: config/routes.rb [hunk 2]
```diff
@@ -28,5 +28,6 @@
   scope module: :system_admin, path: "system-admin" do
     resources :applicants, only: %i[index show edit update]
     get "/dashboard", to: "dashboard#show"
+    resources "reports"
```
Comment (raul-gracia) [inline config/routes.rb:31]: we've only using index and show actions, we should restrict this with the :only keyword.
```suggestion
    resources "reports", only: [:show, :index]
```

**Review by raul-gracia — APPROVED** [review]

File: app/models/reports/home_office.rb (hunk 1 above)
Comment (padv2010) [reply app/models/reports/home_office.rb]: Addressed in: https://github.com/DFE-Digital/get-a-teacher-relocation-payment/pull/144/commits/3dbdaa548460a8382c2b8851200ada98b30b64e2

**Pull request merged**